src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -252,19 +274,109 @@ void Engine::ApplyPerformanceHints(std::map<std::string, std::string> &config, c
+            config[CONFIG_KEY_INTERNAL(SMALL_CORE_STREAMS)] = std::to_string(tput_hints.second.small_core_streams);
+            config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_BIG)] =
+                std::to_string(tput_hints.second.threads_per_stream_big);
+            config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_SMALL)] =
+                std::to_string(tput_hints.second.threads_per_stream_small);
+            config[CONFIG_KEY_INTERNAL(SMALL_CORE_OFFSET)] = std::to_string(tput_hints.second.small_core_offset);
+        }
+    }
+}
+
+int Engine::GetModelPreferThreads(const std::shared_ptr<ngraph::Function>& ngraphFunc) const {
+    const auto isa = dnnl::get_effective_cpu_isa();
+    float isaSpecificThreshold = 1.0f;
+    switch (isa) {
+    case dnnl::cpu_isa::sse41:
+        isaSpecificThreshold = 0.5f;
+        break;
+    case dnnl::cpu_isa::avx2:
+    case dnnl::cpu_isa::avx512_core:
+        isaSpecificThreshold = 1.0f;
+        break;
+    case dnnl::cpu_isa::avx512_core_vnni:
+    case dnnl::cpu_isa::avx2_vnni:
+        isaSpecificThreshold = 2.0f;
+        break;
+    case dnnl::cpu_isa::avx512_core_amx:
+        isaSpecificThreshold = 4.0f;
+        break;
+    default:
+        isaSpecificThreshold = 1.0f;
+    }
+    // the more "capable" the CPU in general, the more streams we may want to keep to keep it utilized
+    const float memThresholdAssumeLimitedForISA = ov::MemBandwidthPressure::LIMITED / isaSpecificThreshold;
+    const float L2_cache_size = dnnl::utils::get_cache_size(2 /*level*/, true /*per core */);
+    ov::MemBandwidthPressure networkToleranceForLowCache =
+        ov::MemBandwidthPressureTolerance(ngraphFunc, L2_cache_size, memThresholdAssumeLimitedForISA);
+    auto model_prefer = IStreamsExecutor::Config::StreamMode::DEFAULT;
+    if (networkToleranceForLowCache.max_mem_tolerance == ov::MemBandwidthPressure::UNKNOWN) {
+        if ((networkToleranceForLowCache.ratio_compute_convs == ov::MemBandwidthPressure::ALL) ||
+            (networkToleranceForLowCache.ratio_compute_deconvs == ov::MemBandwidthPressure::ALL)) {
+            // all relevant layers (convs, etc) are compute-limited, the most aggressive val for #streams
+            model_prefer = IStreamsExecutor::Config::StreamMode::AGGRESSIVE;
+        }  // otherwise (no recognized layers) falling back to the default value
+    } else if (networkToleranceForLowCache.max_mem_tolerance > memThresholdAssumeLimitedForISA) {
+        // network is below the ISA-specific threshold
+        model_prefer = IStreamsExecutor::Config::StreamMode::AGGRESSIVE;
+    } else if (networkToleranceForLowCache.max_mem_tolerance > ov::MemBandwidthPressure::LIMITED) {
+        // network is below general threshold
+        model_prefer = IStreamsExecutor::Config::StreamMode::LESSAGGRESSIVE;
+    }
+    return model_prefer;
+}
+
+Engine::StreamCfg Engine::ParseStreamsTable(std::vector<std::vector<int>> streams_table) const {
```

Comment:
Could we parse `streams_info_table` to config key directly?

Reply:
I can't get the key directly from `streams_info_table` due to its structure.